### PR TITLE
avoid sending an email alert on Google 503 errors

### PR DESCRIPTION
**Pull request by briskt**
### Changed
- Do not send an email alert for Google API 503 response

### CODE REVIEW

File: sync.go [hunk 1]
```diff
@@ -80,44 +80,50 @@ func RunSync(configFile string) error {
...
 
 		// Apply SyncSet configs (excluding source/destination as appropriate)
-		err = source.ForSet(syncSet.Source)
-		if err != nil {
-			msg := fmt.Sprintf(`Error setting source set on syncSet "%s": %s`, syncSet.Name, err)
-			syncSetLogger.Println(msg)
-			errors = append(errors, msg)
+		if err = source.ForSet(syncSet.Source); err != nil {
+			err = fmt.Errorf(`Error setting source set on syncSet "%s": %w`, syncSet.Name, err)
+			syncSetLogger.Println(err)
```
Comment (jason-jackson) [inline sync.go:98]: I'm thinking this should be moved to inside `handleSyncError`, but then you'd either need to pass `syncSetLogger` or change it to `log.Println()`.  Personally, I don't see any issues with using `log.Println` as the output and Flags for log are set at the beginning of `RunSync()`, or is there a reason we're using `log.New()`?

**Review by jason-jackson — APPROVED** [review]

File: sync.go (hunk 1 above)
Comment (briskt) [reply sync.go]: The `syncSetLogger` is customized, so if it were to use simply `log.Println` it wouldn't use the custom prefix. Or maybe you're suggesting that for each pass through the loop, the global logger gets redefined. I suppose that would work, but I haven't tried it.

File: sync.go [hunk 2]
```diff
@@ -80,44 +80,50 @@ func RunSync(configFile string) error {
...
+		alert.SendEmail(config.Alert, fmt.Sprintf("Sync error(s):\n%s", strings.Join(alertList, "\n")))
 	}
 
 	log.Printf("Personnel sync completed at %s", time.Now().UTC().Format(time.RFC1123Z))
 	return nil
 }
+
+func handleSyncError(err error, alertList []string) []string {
+	var syncError internal.SyncError
+	if isSyncError := errors.As(err, &syncError); !isSyncError || (isSyncError && syncError.SendAlert) {
```
[the remark this answers is not in the document]
Comment (briskt) [reply sync.go]: Thanks.

File: sync.go (hunk 1 above)
Comment (jason-jackson) [reply sync.go]: Hmm, what about setting the prefix each loop then reverting at end?  That said current implementation is good too.

**Pull request merged**